### PR TITLE
Query Pagination: do not render pagination container if there's no pagination

### PR DESCRIPTION
**Pull request by jffng**
## Description
This PR adds a render callback to the query pagination block. This way, the pagination container element is only rendered if pagination exists.

I think rendering an empty container in most cases is harmless, but is it better to actually not add anything to the DOM if there isn't content present? There are probably other cases where this happens in FSE blocks..

This is an attempt to solve https://github.com/Automattic/themes/issues/4268 where we style the wrapper only when pagination is present.

## How has this been tested?
1. Create a post or index template with the following query block markup on a site with fewer than five posts:

```
<!-- wp:query {"tagName":"main","layout":{"inherit":true},"queryId":1,"query":{"perPage":5,"offset":0,"postType":"post","categoryIds":[],"tagIds":[],"order":"desc","orderBy":"date","author":"","search":"","sticky":"","inherit":true}} -->
<main class="wp-block-query">
	<!-- wp:post-template -->
		<!-- wp:post-date /-->
		<!-- wp:post-title /-->
		<!-- wp:post-featured-image /-->
		<!-- wp:post-excerpt /-->
	<!-- /wp:post-template -->
	<!-- wp:query-pagination {"align":"wide"} -->
	<!-- wp:query-pagination-previous /-->
	<!-- wp:query-pagination-numbers /-->
	<!-- wp:query-pagination-next /-->
	<!-- /wp:query-pagination -->
</main>
<!-- /wp:query -->
``` 

2. Verify that no pagination container is rendered 
3. Verify relevant block attributes ("wide" in the above example) are present on the wrapper

## Types of changes
Bug fix? 

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/query-pagination/index.php [hunk 1]
```diff
@@ -5,12 +5,36 @@
...
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ *
+ * @return string Returns the wrapper for the Query pagination.
+ */
+function render_block_core_query_pagination( $attributes, $content ) {
+	$wrapper_attributes = get_block_wrapper_attributes();
+
+	if ( empty( $content ) || ctype_space( $content ) ) {
```
Comment (jffng) [inline packages/block-library/src/query-pagination/index.php:19]: For some reason the `$content` is a long string of whitespace when it's empty, so I added the `ctype_space` condition.

**Comment by scruffian** [comment]
The problem I am seeing with this is that when the block has already been saved it contains content like this:

```<div class="wp-block-query-pagination alignwide"> </div>```

Do we need to check for $content at all, or can we always just do a server rendering?

**Comment by jffng** [comment]
I see what you mean. But if we don't check $content, how else could we avoid rendering an empty container?

**Comment by scruffian** [comment]
Hmm. I guess the alternative is to check for `<div class="wp-block-query-pagination alignwide"> </div>`

**Comment by scruffian** [comment]
Thinking about this some more, I think it's good. If won't deal with blocks that have already been saved, but it will work for new blocks.

**Review by scruffian — APPROVED** [review]

**Review by Mamaduka — CHANGES_REQUESTED** [review]
Hi, @jffng

I think this change will require deprecation as well and maybe test/snapshot updates.

PR also fixes #30522.

**Comment by jffng** [comment]
Closing in favor of #35092

**Pull request closed without merging**